src/main/java/com/imgix/URLHelper.java

```diff
@@ -192,11 +201,29 @@ public static String decodeURIComponent(String s) {
     String result = null;
 
     try {
-      result = URLDecoder.decode(s, "UTF-8");
+      result = URLDecoder.decode(s, UTF_8);
     } catch (UnsupportedEncodingException e) {
       result = s;
     }
 
     return result;
   }
+
+  public static String sanatizePath(String path) {
+    // Strip leading slash first (we'll re-add after encoding)
+    path = path.replaceAll("^/", "");
+
+    if (path.startsWith("http://") || path.startsWith("https://")) {
+      // Use encodeURIComponent to ensure *all* characters are handled,
+      // since it's being used as a path
+      return "/" + URLHelper.encodeURIComponent(path);
+    } else if (path.startsWith("http%3A") || path.startsWith("https%3A")) {
+      // To nothing to the URL being used as path since already encoded
+      return "/" + path;
+    } else {
```

Comment:
I like the idea here, but I fear that actively supporting some pre-encoded cases is going to create more problems for us down the road. We tell[1] our users to pass in any inputs un-encoded, so I think we need to discuss it as a team if we're ever going to diverge from that. 

[1] not as often as we should apparently; see js-core's [docs](https://github.com/imgix/js-core/blob/main/README.md#L53)
[2] [js-core's sanitizePath](https://github.com/imgix/js-core/blob/main/src/index.js#L96-L104) method for reference

Reply:
On second thought, this might be here because of [this block](https://github.com/imgix/imgix-java/pull/68/files#diff-f0261780cd53d140fe700704b8126e6fd2e47de88f3c5f95518324aabc125cf2L99-L102), in which case we'll have to leave it for now.

Reply:
@luqven Sorry, I didn't actually form my thought as a question here but is my last comment true? In other words, should we resolve this issue and move on?

Reply:
Ah i see.

In part yes and in part no. This was also based off of a discussion with Eric about potentially handling encoded URLS.

I'm happy to remove this conditional for now so we can discuss this some more. Makes the most sense now that I think about it.

Reply:
Seems that the functionality in the [aforementioned block](https://github.com/imgix/imgix-java/pull/68/files#diff-f0261780cd53d140fe700704b8126e6fd2e47de88f3c5f95518324aabc125cf2L99-L102) comes from [this commit](https://github.com/imgix/imgix-java/pull/21/commits/e898cc123dff2cad1ed05092c4fd94d07ee23c06) in 2017. Makes me think its safe to remove this decoding behavior altogether for now.

Reply:
As mentioned in [this comment ](https://github.com/imgix/imgix-java/pull/70#issuecomment-813405830) on #70 , we need to allow encoded proxy paths.